fragment.go

```diff
@@ -1325,71 +1325,65 @@ func (f *fragment) mergeBlock(id int, data []pairSet) (sets, clears []pairSet, e
 // bulkImport bulk imports a set of bits and then snapshots the storage.
 // This does not affect the fragment's cache.
 func (f *fragment) bulkImport(rowIDs, columnIDs []uint64) error {
-	f.mu.Lock()
-	defer f.mu.Unlock()
 	// Verify that there are an equal number of row ids and column ids.
 	if len(rowIDs) != len(columnIDs) {
 		return fmt.Errorf("mismatch of row/column len: %d != %d", len(rowIDs), len(columnIDs))
 	}
 
 	// Disconnect op writer so we don't append updates.
-	f.storage.OpWriter = nil
-
+	localBitmap := roaring.NewBitmap()
```

Comment:
why all these changes to the `bulkImport` method? is this related to the roaring import work?

Reply:
yes, there was a fundamental change to how the snapshot method works, basically a local bitmap is built, then merged into the existing shard bitmap as opposed to setting bits on the larger file.